Sources/BaseButtonBarPagerTabStripViewController.swift

```diff
@@ -0,0 +1,647 @@
+            return minimumCellWidths
+        } else {
+            let stretchedCellWidthIfAllEqual = (collectionViewAvailableVisibleWidth - cellSpacingTotal) / CGFloat(numberOfCells)
+            let generalMinimumCellWidth = calculateStretchedCellWidths(minimumCellWidths, suggestedStretchedCellWidth: stretchedCellWidthIfAllEqual, previousNumberOfLargeCells: 0)
+            var stretchedCellWidths = [CGFloat]()
+
+            for minimumCellWidthValue in minimumCellWidths {
+                let cellWidth = (minimumCellWidthValue > generalMinimumCellWidth) ? minimumCellWidthValue : generalMinimumCellWidth
+                stretchedCellWidths.append(cellWidth)
+            }
+
+            return stretchedCellWidths
+        }
+    }
+
+    private var shouldUpdateButtonBarView = true
+}
+
+// MARK: Protocols
+
+public protocol IndicatorInfoProvider {
+
+    func indicatorInfo(for pagerTabStripController: PagerTabStripViewController) -> IndicatorInfo
+
+}
+
+public protocol PagerTabStripIsProgressiveDelegate: class {
+
+    func updateIndicator(for viewController: PagerTabStripViewController, fromIndex: Int, toIndex: Int, withProgressPercentage progressPercentage: CGFloat, indexWasChanged: Bool)
+}
+
+public protocol PagerTabStripDataSource: class {
+
+    func viewControllers(for pagerTabStripController: PagerTabStripViewController) -> [UIViewController]
+}
+
+// MARK: PagerTabStripViewController
+
+open class PagerTabStripViewController: UIViewController, UIScrollViewDelegate {
+
+    @IBOutlet weak public var containerView: UIScrollView!
+
+    open weak var delegate: PagerTabStripIsProgressiveDelegate?
+    open weak var datasource: PagerTabStripDataSource?
+
+    open private(set) var viewControllers = [UIViewController]()
+    open private(set) var currentIndex = 0
+    open private(set) var preCurrentIndex = 0 // used *only* to store the index to which move when the pager becomes visible
+
+    open var pageWidth: CGFloat {
+        return containerView.bounds.width
+    }
+
+    open var scrollPercentage: CGFloat {
+        if swipeDirection != .right {
+            let module = fmod(containerView.contentOffset.x, pageWidth)
+            return module == 0.0 ? 1.0 : module / pageWidth
+        }
+        return 1 - fmod(containerView.contentOffset.x >= 0 ? containerView.contentOffset.x : pageWidth + containerView.contentOffset.x, pageWidth) / pageWidth
+    }
+
+    open var swipeDirection: SwipeDirection {
+        if containerView.contentOffset.x > lastContentOffset {
+            return .left
+        } else if containerView.contentOffset.x < lastContentOffset {
+            return .right
+        }
+        return .none
+    }
+
+    override open func viewDidLoad() {
+        super.viewDidLoad()
+        let containerViewAux = containerView ?? {
+            return UIScrollView(frame: CGRect(x: 0, y: 0, width: view.bounds.width, height: view.bounds.height))
+            }()
+        containerView = containerViewAux
```

Comment:
somehow this style doesn't mix well with the rest of the swift code. Can you adapt that to fit in more with the rest?